src/runtime/importhook.cs

```diff
@@ -9,13 +8,46 @@ namespace Python.Runtime
     /// </summary>
     internal static class ImportHook
     {
-        private static IntPtr py_import;
         private static CLRModule root;
-        private static MethodWrapper hook;
         private static IntPtr py_clr_module;
 
         private static IntPtr module_def = IntPtr.Zero;
 
+        private const string LoaderCode = @"
+import importlib.abc
+import sys
+
+class DotNetLoader(importlib.abc.Loader):
+
+    def __init__(self):
+        super(DotNetLoader, self).__init__()
+
+    @classmethod
+    def exec_module(klass, mod):
+        # This method needs to exist.
+        pass
+
+    @classmethod
+    def create_module(klass, spec):
+        import clr
+        return clr._LoadClrModule(spec)
+
+class DotNetFinder(importlib.abc.MetaPathFinder):
+    
+    def __init__(self):
+        super(DotNetFinder, self).__init__()
+    
+    @classmethod
+    def find_spec(klass, fullname, paths=None, target=None): 
+        import clr
+        if (hasattr(clr, '_availableNamespaces') and fullname in clr._availableNamespaces):
```

Comment:
Why is this written so defensively?

Reply:
As things are right now, when the domain unloads, the attribute is removed.